src/Compilers/CSharp/Test/Syntax/LexicalAndXml/XmlDocCommentTests.cs

```diff
@@ -30,6 +31,36 @@ private SyntaxTree Parse(string text, params string[] defines)
             return SyntaxFactory.ParseSyntaxTree(itext, options);
         }
 
+        [Fact]
+        public void DocCommentWriteException()
+        {
+            var comp = CreateCompilationWithMscorlib(@"
+/// <summary>
+/// Doc comment for <see href=""C"" />
+/// </summary>
+public class C
+{
+    /// <summary>
+    /// Doc comment for method M
+    /// </summary>
+    public void M() { }
+}");
+            var diags = new DiagnosticBag();
+            var badStream = new BrokenStream();
+            badStream.BreakHow = BrokenStream.BreakHowType.ThrowOnWrite;
+
+            DocumentationCommentCompiler.WriteDocumentationCommentXml(
+                comp,
+                null,
+                badStream,
+                diags,
+                default(CancellationToken));
+
+            diags.Verify(
+                // error CS1569: Error writing to XML documentation file: I/O error occurred.
+                Diagnostic(ErrorCode.ERR_DocFileGen).WithArguments("I/O error occurred.").WithLocation(1, 1));
```

Comment:
@agocke `WithLocation(1, 1)` looks like the first character in the source rather than `Location.None`.